### DISCUSSION TITLE
JS SDK authStore.exportToCookie() encodes user info into cookie value?

### DISCUSSION THREAD

**Discussion by YoRHa-A5** [pocketbase/pocketbase#2718] [Q&A] [answered]
I did not notice this before, so I just wanted to ask if this is the normal behavior.
The value of the pb_auth cookie seems to included user data, here is some of the encoded and formatted data:

```json
{
   "token":"ey...",
   "model":{
      "about":"some text about user",
      "collectionId":"systemprofiles0",
      "collectionName":"users",
      "created":"2022-10-06 18:26:15.560Z",
      "email":"user@123.com",
      "emailVisibility":false,
      "id":"123",
      "onboarded":true,
      "pictures":[
         "123.jpg",
         "456.jpg",
         "789.jpg"
      ],
      "roles":[
         
      ],
      "updated":"2023-06-14 16:29:14.787Z",
      "username":"user1",
      "verified":true,
      "expand":{
         
      }
   }
}
```
Like I said, I did not see this before, so just wanted to ask whether that's fine having URL encoded user data inside the cookie value? 🤔

**Comment by ganigeorgiev** [comment, marked as answer]
The cookie is used just as an optional transport mechanism between the browser and your node server (the same way as we store the auth state by default in `localStorage`). The cookie is not encrypted so you shouldn't share it with 3rd party applications (by default it is exported with `SameSite=Strict`). If you don't want the user data, you can create your own cookie (`exportToCookie` and `loadFromCookie` are just helpers and not required to be used).

Keep in mind that if your node application is performing some permission check based on the cookie token or user data you need to first validate the loaded authStore against PocketBase. This can be done by calling `authRefresh()` for your auth collection. For an example SSR integration flow, you can check https://github.com/pocketbase/js-sdk#ssr-integration.
